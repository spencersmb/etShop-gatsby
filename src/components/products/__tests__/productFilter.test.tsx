import ProductFilter from '@components/products/productFilter'
import React from 'react'
import renderer from 'react-test-renderer'
import { render } from 'react-testing-library'

const props = {
	handleClick: jest.fn(),
	filter: ''
}

describe('product Filter', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<ProductFilter {...props}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should handleClick correctly', () => {
		const modalRender = render(<ProductFilter {...props} filter=''/>)
		const button = modalRender.getAllByTestId('filterItems')[0]
		button.click()
		expect(props.handleClick).toHaveBeenCalledTimes(1)
		expect(props.handleClick).toHaveBeenCalledWith('fonts')
	})
})