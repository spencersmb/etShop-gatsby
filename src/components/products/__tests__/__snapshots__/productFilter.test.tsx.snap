// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product Filter renders correctly 1`] = `
<div
  className="sc-bdVaJa gjJKSz"
>
  <div
    className="sc-bwzfXH cRmAIp"
  >
    <div
      className="sc-htpNat ezCnHF"
    >
      <div
        className="sc-gzVnrw ijTmdr"
      >
        <span>
          <test-file-stub />
        </span>
        <h4>
          Filter
        </h4>
      </div>
      <ul>
        <li
          className="sc-bxivhb iGBIdr"
          data-filtertype="fonts"
          data-testid="filterItems"
          onClick={[Function]}
        >
          <div
            className="sc-ifAKCX hSMeWm"
          >
            <span
              className="sc-EHOje hsbIlr"
            >
              <test-file-stub />
            </span>
            Fonts
          </div>
          <span
            className="slider sc-bZQynM hmpjWr"
          />
        </li>
        <li
          className="sc-bxivhb iGBIdr"
          data-filtertype="textures"
          data-testid="filterItems"
          onClick={[Function]}
        >
          <div
            className="sc-ifAKCX hSMeWm"
          >
            <span
              className="sc-EHOje hsbIlr"
            >
              <test-file-stub />
            </span>
            Textures
          </div>
          <span
            className="slider sc-bZQynM hmpjWr"
          />
        </li>
        <li
          className="sc-bxivhb iGBIdr"
          data-filtertype="templates"
          data-testid="filterItems"
          onClick={[Function]}
        >
          <div
            className="sc-ifAKCX hSMeWm"
          >
            <span
              className="sc-EHOje hsbIlr"
            >
              <test-file-stub />
            </span>
            Templates
          </div>
          <span
            className="slider sc-bZQynM hmpjWr"
          />
        </li>
      </ul>
      <div
        className="sc-htoDjs bwflZR"
        data-filtertype=""
        data-testid="filterItems"
        onClick={[Function]}
      >
        View all
      </div>
    </div>
  </div>
</div>
`;
